src/MessageReader.js

```diff
@@ -285,15 +289,16 @@ const createParser = (types: RosMsgDefinition[], freeze: boolean) => {
     const reader = new StandardTypeReader(buffer);
     return _read(reader);
   };
-};
+}
 
 export class MessageReader {
+  parserCode: string;
   reader: (buffer: Buffer) => any;
 
   // takes an object message definition and returns
   // a message reader which can be used to read messages based
   // on the message definition
-  constructor(definitions: RosMsgDefinition[], options: { freeze?: ?boolean } = {}) {
+  constructor(definitions: RosMsgDefinition[], options: { freeze?: ?boolean, parserCode?: ?string } = {}) {
```

Comment:
Instead of overloading the constructor, what if we used a static factory function for creating `MessageReaders` from `parserCode`? Then we could avoid the case where `options.freeze` and `options.parserCode` are both set
```
const reader1 = new MessageReader(messageDefinition);
const myReader =  MessageReader.fromParserCode(reader1.parserCode);
```

Reply:
@hassoncs I originally went that direction, but I think I'd still need to overload the constructor - I couldn't figure out a way to create a new object with a class type without doing that. Another option would be to no longer make this a class but instead just a normal `createMessageReader` function.